packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php

```diff
@@ -0,0 +1,358 @@
+    }
+
+    /**
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterOptions
+     */
+    protected function createProductFilterOptionsInstance(): ProductFilterOptions
+    {
+        return new ProductFilterOptions();
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Flag\Flag $flag
+     * @param int $count
+     * @param bool $isAbsolute
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\FlagFilterOption
+     */
+    protected function createFlagFilterOption(Flag $flag, int $count, bool $isAbsolute): FlagFilterOption
+    {
+        return new FlagFilterOption($flag, $count, $isAbsolute);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\Brand $brand
+     * @param int $count
+     * @param bool $isAbsolute
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\BrandFilterOption
+     */
+    protected function createBrandFilterOption(Brand $brand, int $count, bool $isAbsolute): BrandFilterOption
+    {
+        return new BrandFilterOption($brand, $count, $isAbsolute);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter $parameter
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ParameterValueFilterOption[] $parameterValueFilterOptions
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ParameterFilterOption
+     */
+    protected function createParameterFilterOption(Parameter $parameter, array $parameterValueFilterOptions): ParameterFilterOption
+    {
+        return new ParameterFilterOption($parameter, $parameterValueFilterOptions);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue $brand
+     * @param int $count
+     * @param bool $isAbsolute
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ParameterValueFilterOption
+     */
+    protected function createParameterValueFilterOption(ParameterValue $brand, int $count, bool $isAbsolute): ParameterValueFilterOption
+    {
+        return new ParameterValueFilterOption($brand, $count, $isAbsolute);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfig $productFilterConfig
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterCountData $productFilterCountData
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterOptions
+     */
+    protected function createProductFilterOptions(
+        ProductFilterConfig $productFilterConfig,
+        ProductFilterCountData $productFilterCountData,
+        ProductFilterData $productFilterData
+    ): ProductFilterOptions {
+        $productFilterOptions = $this->createProductFilterOptionsInstance();
+        $productFilterOptions->minimalPrice = $productFilterConfig->getPriceRange()->getMinimalPrice();
```

Comment:
When I disable the module "Number of products in filter" for following query I get `internal server error` with debug message `Cannot return null for non-nullable field ProductFilterOptions.minimalPrice.`

```
{
  category(uuid: "fca28f54-e568-4b3a-b6f6-a2d0ce3e8691") { ### TV-Audio
    name
    products(
      first: 10
    ) {
      totalCount
      productFilterOptions {
        minimalPrice
        maximalPrice
        brands{
          brand {
            name
          }
          count
          isAbsolute
        }
        flags{
          flag {
            name
          }
          count
          isAbsolute
        }
        inStock
        parameters{      
          name
          values{
            text
          }
        }
      }
      edges {
        node {
          name
        }
      }
    }
    
  }
}
```

Can you please check it?

Reply:
Nice catch! e3f432320e37ddd027d8e7b0804fdc87c9da9c4c